jdbc/src/main/java/com/interface21/jdbc/core/PreparedStatementSetter.java

```diff
@@ -0,0 +1,19 @@
+package com.interface21.jdbc.core;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public class PreparedStatementSetter {
+
+    private final Object[] args;
+
+    public PreparedStatementSetter(final Object[] args) {
+        this.args = args;
+    }
+
+    public void setValues(final PreparedStatement pstmt) throws SQLException {
+        for (int i = 0; i < args.length; i++) {
+            pstmt.setObject(i + 1, args[i]);
```

Comment:
args[i]가 null인 경우는 어떻게 될까요? 이 pss가 해결하지 못하는 종류의 입력도 있을까요?

Reply:
[반영] NPE가 발생하겠네요..! PreparedStatementSetter 생성 시 null인 args로는 생성하지 못하도록 검증 로직을 추가했습니다

Reply:
현재 코드는 args[i]가 아닌 args가 null일때 NPE를 발생시키도록 하는 것으로 보여요.
args[i]가 null일 때 setObject를 수행하면 어떤 일이 일어나는 지 아시면 좋을 것 같습니다.
>Note: Not all databases allow for a non-typed Null to be sent to the backend. For maximum portability, the setNull or the setObject(int parameterIndex, Object x, int sqlType) method should be used instead of setObject(int parameterIndex, Object x).

Reply:
NPE가 그냥 발생하는 것과 코드에서 NPE를 명시해주는 것은 어떤 차이가 있을까요?